taichi/ir/frontend_ir.cpp

```diff
@@ -1332,6 +1310,14 @@ Expr ASTBuilder::expr_alloca() {
   return var;
 }
 
+Expr ASTBuilder::insert_func_call(Function *func, const ExprGroup &args) {
+  auto var = Expr(std::make_shared<IdExpression>(get_next_id()));
```

Comment:
Are functions guaranteed to return a value?

Reply:
If it has no return value, it returns an empty struct.

Reply:
This feels iffy... Some IR might not like the idea of an empty struct

Reply:
Maybe we can deal with this in the codegen. We can return void in the codegen if there is no return value (the return type is an empty struct in CHI-IR). In this case the return value is not used anywhere so we don't need to care about its value.

Reply:
Maybe it's better to return a std::optional